buddyboss/buddyboss-platform
# Sync Activity Comments > Update user timeline for Blog, CPTs and Forums auto-generated activity.

### All Submissions:






### Changes proposed in this Pull Request:



Fixes #2220 

### Theme PR
https://github.com/buddyboss/buddyboss-theme/pull/1380

### How to test the changes in this Pull Request:

- Rework on Activity Settings:
	
	- Remove existing settings to Sync comment with Blogs, CPT, and Forums.

	- Provide new checkbox settings to Enable comments YES/NO. Follow the settings wireframes in the video shared.

	- Make sure this is backward compatible. IF customer have "Sync Comment" settings enabled already then
	new setting Enable comments should be checked as well for Blog Posts and CPTs.

- For Forums Discussion and Replies auto-generated activity in the frontend (When activity settings enabled in the backend)
	
	- Take the "discussion" name out of the activity title and put it underneath for both Discussion and Replies. Follow the activity feed wireframe from the video.

	- Make sure for long text we have the option for 'read more'. We already have that working on Activity Content and Blog post.
	
	- Do not provide the option to comment on auto-generated activity for Discussion and Replies.

	- Provide a new link "Join Discussion" for both Discussion and Reply activity which will redirect to dedicated discussion and reply.

	- Provide new link "Quick Reply" for Discussion activity only.

	- Quick reply will allow responding to Discussion right from the Activity screen.

		- Show the exact same popup as the reply form. Follow popup wireframes in the video shared.

	- For backward compatibility: Since Forum Discussion and Reply auto-generated activity comments are not getting synced at the moment.
	We have existing customers of an auto-generated activity received comments.
	So we need to add logic for an existing activity to check if a comment exists then show but do not allow to add more activity comments.
	For new Activity anyways comments will not show.

- For Blogs and CPTs activity in the Frontend > When activity settings enabled in the backend

	- Instead of loading activity comments and syncing with WordPress Comments. Simply show WordPress comments directly here. 
	Blog post comment will show up here as it is.

	- Provide option add WP comment and reply from this screen.

	- Follow all rules from single post/CPT settings.
	If comments disabled in the Settings OR post comment disabled then the "Comment" button will not show.
	There are instances we don't show comments at all. Like for WooCommerce Product. There may be other instances as well.
	Check logic and NOTE while sharing a proof on all instances we will not show an option for comments.
	Need to make sure that logic works as it is.

### Proof Screenshots or Video

- Rework on Activity Settings:
https://www.loom.com/share/e980cbfee937434984dbccc677c7bde6

- For Forums Discussion and Replies auto-generated activity in the frontend (When activity settings enabled in the backend)
https://www.loom.com/share/e9ce458e5dbf428fa8a581725f66fdab

- Instead of loading activity comments and syncing with WordPress Comments. Simply show WordPress comments.
https://www.loom.com/share/07434e58703a4e57ac5312acd9e747fc

- Activity blog post comment depth or nested level of comments.
https://www.loom.com/share/8383bf052cf64f908546a13aa963cedd

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
